homeassistant/components/weenect/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""The Weenect ConfigFlow."""
+import logging
+
+from aioweenect import AioWeenect
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class WeenectFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for weenect."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        self._errors = {}
+
+        entries = self._async_current_entries()
+        for entry in entries:
+            if (
+                entry.data[CONF_USERNAME] == user_input[CONF_USERNAME]
+                and entry.data[CONF_PASSWORD] == user_input[CONF_PASSWORD]
```

Comment:
By the looks of it, `user_input` could be `None` at this point (or if it couldn't, why bother checking on line 39)?

If so, this could raise a KeyError.